### ISSUE TITLE
[Lab3-3] Why (de)conv followed by (de)conv ?

### ISSUE THREAD

**Issue by doom8199** [2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN#14]
Dear TA:
    In lab3-3 's requirement, I found that there are this structure that (de)conv followed by (de)conv in the three networks.  Why not follow by BatchNorm or ReLU ? Do any documents of theoretical or empirical results explain for that ?
    I think concatenate convolution operators can be equivalent to exactly one convolution operator.

**Comment by hui-po-wang** [comment]
Hi @doom8199 ,

Yes, you're right.

Indeed, the task assigned to me is to translated a file from tensorflow to pytorch and the file is implemented by another Prof. Chiu's student (I am not his student). Maybe he missed something, and I didn't point it out, too.

However, I've tested the architecture and I am sure it works, so please follow the architecture for now (you can also try to use only one conv. ). I will check the architecture again, add non-linearity into the model, and measure the computing time (I  guess it would take more time, maybe it's better to remain the current architecture because of the computation power of your machine).

Thanks a lot for pointing out this interesting problem, I'll share results with you as soon as possible.

**Comment by doom8199** [comment]
Dear TA:
    Thanks for the quickly apply. I will also try other modification (such as one conv, or insert non-linear layers between convs) after complete HW3-3's requirements.

**Comment by a514514772** [comment]
Hi @doom8199,

I

**Comment by hui-po-wang** [comment]
Hi @doom8199 ,

I think the original implement still makes sense because we have not only convolution operations but also down-sampling operations (stride is not equal to 1).

**Comment by mryzlee** [comment]
Hi a514514772,

Actually, down-sampling operation (stride=2) is also a kind of linear operation.
You can regard stride=2 down-sampling as a 1-to-1 direct connection layer with even samples weighting=1 and odd samples weighting=0.
So, I think it can still be merged.
